protocol/superchainerc20/L1-native-factory.md

```diff
@@ -0,0 +1,170 @@
+        type(BeaconProxy).creationCode,
+        abi.encode(Predeploy.OptimismSuperchainERC20Beacon, abi.encodeCall(SUPERCHAIN_IMPL.initialize.selector, (_remoteToken, _name, _symbol, _decimals)))
+    );
+
+    bytes32 salt = keccak256(abi.encode(_remoteToken, _name, _symbol, _decimals));
+    _superchainERC20 = CREATE3.deploy(salt, _creationCode);
+
+    deployments[_superchainERC20] = _remoteToken;
+
+    emit OptimismSuperchainERC20Deployed(_superchainERC20, _remoteToken _name, _symbol, _decimals);
+  }
+}
+```
+
+The name `_remoteToken` was chosen over `_l1Token` to contemplate future L2 native cases.
+
+**BeaconProxy (OptimismSuperchainERC20)**
+
+The Token should follow a simple BeaconProxy implementation, like Openzeppelin:
+
+https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/proxy/beacon/BeaconProxy.sol
+
+**Beacon Contract**
+
+```solidity
+contract OptimismSuperchainERC20Beacon is IBeacon {
+  /// TODO: Replace with real implementation address
+  address internal constant IMPLEMENTATION_ADDRESS = 0x0000000000000000000000000000000000000000;
+
+  function implementation() external pure override returns (address) {
+    return IMPLEMENTATION_ADDRESS;
+  }
+}
+```
+
+**Implementation**
+The implementation should include
+
+- An `initialize` function that takes `(address _remoteToken, string _name, string _symbol, uint8 decimals)` and stores these in the storage of the BeaconProxy.
+- A function to reinitialize after proxy deployments or upgrades.
+
+## User Stories
+
+### `OptimismSuperchainERC20` deployment
+
+1. Anyone can deploy a `OptimismSuperchainERC20` by calling the `deploy` function in the `OptimismSuperchainERC20Factory` contract with `_remoteToken`, `_name`, `_symbol` and `decimals` as inputs.
+2. The factory stores the `_remoteToken` address for each deployed SuperchainERC20 address.
+
+```mermaid
+sequenceDiagram
+  participant Alice
+  participant FactoryProxy
+  participant FactoryImpl
+  participant CREATE3Proxy
+  participant BeaconProxy as OptimismSuperchainERC20 BeaconProxy
+  participant Beacon Contract
+  participant Implementation
+
+  Alice->>FactoryProxy: deploy(remoteToken, name, symbol, decimals)
+  FactoryProxy->>FactoryImpl: delegatecall()
+  FactoryProxy-->FactoryProxy: deployments[superchainERC20]=remoteToken
+  FactoryProxy->>CREATE3Proxy: create2()
+  CREATE3Proxy->>BeaconProxy: create()
+  BeaconProxy-->Beacon Contract: reads implementation()
+  BeaconProxy->>Implementation: delegatecall()
+```
+
+### `SuperchainERC20` implementation upgrade
+
+1. Anyone can deploy a new implementation on each chain to be upgraded.
+2. Core developers and Foundation will organize a hardfork between the upgraded chains.
+3. The hardfork will update the implementation address on the Beacon Contract.
+
+## Open Questions
+
+- How should different ERC20 implementations be handled? One or multiple implementations per Beacon Contract predeploy?
+- How should native token deployment in L2 behave?
```

Comment:
I think we should leave this out of scope for this particular project and spike out what the best devex/ux is in the future. My guess is that either minting happens on 1 chain or minting happens on any chain and admin comes from 1 chain

Reply:
Agree it's out of scope. Will remove from here.